File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double maxcount = 0;
        double initial = 0;
        if (nums.size() <= k) {
            for (auto num : nums) {
                maxcount += num;
            }
        }

        for (int i = 0; i < k; i++) {
            initial += nums[i];
        }

        maxcount = initial;

        for (int i = k; i < nums.size(); i++) {
            initial = initial - nums[i - k] + nums[i];
            maxcount = max(initial, maxcount);
        }

        return maxcount / k;
    }
};

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        i = 0
        j = 0
        max_sum = -float('inf')
        current_sum = 0
        
        while j < len(nums):
            current_sum += nums[j]

            if (j - i + 1) == k:
                max_sum = max(max_sum, current_sum)
                current_sum -= nums[i]
                i += 1
            j+= 1
        return max_sum / k



